### PR TITLE
Full 1.4 compatibility

### PR DESCRIPTION
**Pull request by dotellie**
Apologies for the commits, they're a little all over the place unfortunately!

This PR should bring this project up to speed with tailwind 1.4 as far as I can tell on their release notes. (https://tailwindcss.com/docs/release-notes/) I started using tailwind like almost literally yesterday so I might have missed something here just like I did with my last PR. Those previous misses have been corrected here though.

I was a bit split on if I should have also added the theming abilities to the new properties but since most of them inherit from previous properties that were already hard-coded, I decided against it. Probably something to think about in the future though along with possibly building up a framework for retrieving keys and setting them in the template.

### CODE REVIEW

**Comment by muhammadsammy** [comment]
Thank you for the PR!

> Apologies for the commits, they're a little all over the place unfortunately!
> 
Not at all, I really appreciate your contributions

> Probably something to think about in the future though along with possibly building up a framework for retrieving keys and setting them in the template.
>
That's a great idea. I made some change on branch `pseudoclass-variants` and I think it will make accomplishing this a bit easier.

File: src/createFile.ts [hunk 1]
```diff
@@ -70,10 +74,15 @@ export function createFileWithGeneratedTypes({ configFilename, outputFilename }:
     const allOpacities = extendedThemeOpacities ? { ...themeOpacities, ...extendedThemeOpacities } : themeOpacities;
     const opacities = Object.keys(allOpacities).map(opacity => `${prefix}opacity-${opacity}`);
 
-    const themeTextOpacities = isEmpty(THEME_CONFIG?.textOpacity) ? allOpacities : THEME_CONFIG?.textOpacity;
-    const extendedThemeTextOpacities = THEME_CONFIG?.extend?.textOpacity;
-    const allTextOpacities = extendedThemeTextOpacities ? { ...themeTextOpacities, ...extendedThemeTextOpacities } : themeTextOpacities;
-    const textOpacities = Object.keys(allTextOpacities).map(opacity => `${prefix}text-opacity-${opacity}`);
+    const getOpacity = (themePropertyName: string, outputNamePrefix: string) => {
+      const opacities = generateOpacities(allOpacities, THEME_CONFIG, themePropertyName);
```
Comment (muhammadsammy) [inline src/createFile.ts:78]: Looks good to me, great work, @dotellie ! 😃
do you think it's better to change this variable names to avoid shadowing?

Comment (muhammadsammy) [inline src/createFile.ts:78]: Do you think it's better to change these variable names to avoid shadowing?

Comment (dotellie) [reply src/createFile.ts]: Done! :+1:

**Pull request merged**